src/Settings.cpp

```diff
@@ -155,3 +177,42 @@ bool Settings::getStartInTray() const
 
     return enable;
 }
+
+void Settings::setAutostart(bool autostart)
+{
+    m_keyFile.set_boolean(GROUP_GENERAL, "autostart", autostart);
+}
+
+bool Settings::getAutostart()
+{
+    auto autostart = false;
+    try
+    {
+        autostart = m_keyFile.get_boolean(GROUP_GENERAL, "autostart");
+    }
+    catch (Glib::KeyFileError const& error)
+    {
+        std::cerr << "Settings: " << error.what() << std::endl;
+    }
+    return autostart;
+}
+
+void Settings::autostart()
+{
+    auto dfPath = getDesktopFilePath();
+    if(dfPath.empty())
+    {
+        std::cerr << "Settings: Failed to find desktop file" << std::endl;
+        return;
+    }
+
+    auto homeDir = std::getenv("HOME");
+    auto gfileDestPath = strcat(homeDir, "/.config/autostart/whatsapp-for-linux.desktop");
+    auto gfileSrc = Gio::File::create_for_path(dfPath);
+    auto gfileDest = Gio::File::create_for_path(gfileDestPath);
+
+    if(Settings::getInstance().getAutostart())
+        gfileSrc->copy(gfileDest);
+    else
+        gfileDest->remove();
```

Comment:
Do the appropriate checks to see if the file is already nonexistent before removing it to prevent unnecessary operation.

Reply:
I also realized that we don't need to keep this setting in the keyfile since the existense of the desktop file already corresponds to the value of it.

Reply:
How to set state of Autostart switch?
```cpp
autostartSwitch->set_state(Settings::getInstance().getAutostart()); // in MainWindow
```
```cpp
    std::string getDesktopFilePath()
    {
        auto possiblePath = std::string("/usr/local/share/applications/whatsapp-for-linux.desktop");
        auto gfilePath = Gio::File::create_for_path(possiblePath);
        if(gfilePath->query_exists())
            return possiblePath;

        possiblePath = "/usr/share/applications/whatsapp-for-linux.desktop";
        gfilePath = Gio::File::create_for_path(possiblePath);
        if(gfilePath->query_exists())
            return possiblePath;

        possiblePath = "/snap/whatsapp-for-linux/current/share/applications/whatsapp-for-linux.desktop";
        gfilePath = Gio::File::create_for_path(possiblePath);

        if(gfilePath->query_exists())
            return possiblePath;

        return {};
    }
 ```

I can move this function body to `getAutostart` and write MainWindow autostartSwitch like this
```cpp
autostartSwitch->set_state(!Settings::getInstance().getAutostart().empty()); 
```

Reply:
I can do something like this.

```cpp
bool Settings::getAutostart()
{
    auto gfileDestPath = CONFIG_DIR + "/autostart/whatsapp-for-linux.desktop";
    auto gfileDesktop = Gio::File::create_for_path(gfileDestPath);
    return gfileDesktop->query_exists();
}
```

Maybe we should move `CONFIG_DIR + "/autostart/whatsapp-for-linux.desktop";` to namespace because its constant and i used it in setAutostart too.